### PR TITLE
#23559 end times

### PR DESCRIPTION
**Pull request by danchr**
This adjusts end times so that the frontend renders them properly, i.e. inclusive rather than exclusive.

The incorrect behaviour:
screenshot-adbee5b2-1b32-410b-85fc-f7a67a7be75c

### CODE REVIEW

File: mora/validator.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def _is_date_range_valid(parent: typing.Union[dict, str],
             if startdate < start:
                 # start is too late!
                 return False
-        elif start != previous_end:
+        elif start not in (previous_end,
+                           previous_end + datetime.timedelta.resolution):
```
Comment (cmoesgaard) [inline mora/validator.py]: Why is this necessary?

File: mora/util.py [hunk 2]
```diff
@@ -118,6 +118,19 @@ def to_lora_time(s):
         return dt.isoformat()
 
 
+def adjust_end_date(dt):
```
[the remark this answers is not in the document]
Comment (cmoesgaard) [reply mora/util.py]: We should also make sure to note that similar related time logic exists in the UI code.

File: mora/util.py [hunk 3]
```diff
@@ -118,6 +118,19 @@ def to_lora_time(s):
...
+    if (
+        dt not in (POSITIVE_INFINITY, NEGATIVE_INFINITY) and
+        dt.time() >= datetime.time(23, 59)
```
Comment (cmoesgaard) [inline mora/util.py]: Remember to adjust to be as specific as possible.

**Comment by danchr** [comment]
The general approach with ε and subtracting something small is just too unwieldy — simply assuming midnight and dealing in dates instead should be much more maintainable.
